scipy/signal/windows/windows.py

```diff
@@ -1620,6 +1620,117 @@ def exponential(M, center=None, tau=1., sym=True):
     return _truncate(w, needs_trunc)
 
 
+def taylor(M, nbar=4, sll=30, norm=True, sym=True):
+    """
+    Return a Taylor window.
+
+    The Taylor window taper function approximates the Dolph-Chebyshev window’s
+    constant sidelobe level for a parameterized number of near-in sidelobes,
+    but then allows a taper beyond [2]_.
+
+    The SAR (synthetic aperature radar) community commonly uses Taylor weighting
+    for image formation processing because it provides strong, selectable sidelobe
+    suppression with minimum broadening of the mainlobe [1]_.
+
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an
+        empty array is returned.
+    nbar : int
+        Number of nearly constant level sidelobes adjacent to the mainlobe.
+    sll : float
+        Desired peak sidelobe level in decibels (dB) relative to the mainlobe. This
+        should be a positive number.
+    norm : boolean
+        When True (default), normalizes the window such that all values are
+        less than or equal to 1.
```

Comment:
So this parameter isn't really associated with the peak level, and in fact when False the peak level should match more closely with what the user inputs. What the parameter does is divide the entire window by the largest value in the window, otherwise it is left as is (that is, it is equal to what the algorithm specifies).

Reply:
Dividing by the largest value suppresses the suppression (sidelobes are larger than desired). Perhaps that should be noted in the parameter?

Reply:
> Dividing by the largest value suppresses the suppression (sidelobes are larger than desired)

Yes but the DC gain is correspondingly shifted, so the side-lobel level *relative to DC gain* is preserved. Perhaps we should

1. Change `sll` description to say that it's relative to the DC level. Then `norm` doesn't affect this meaning.
2. Add "divides by the largest value for odd-length windows, and the value that would occur between the two repeated values for even-length windows" or so here for the `norm` parameter, to make it clear what actually happens.
3. Add something somewhere (probably also under `norm`) saying that `norm=False` will mean that gain at DC is 1 (0 dB) and the side lobes will be `sll` dB down from this.

Reply:
OK, all these suggestions seem very reasonable. I'll go ahead and add them.